lib/haml_lint/reporter/json_reporter.rb

```diff
@@ -0,0 +1,34 @@
+module HamlLint
+  # Outputs JSON
+  class Reporter::JsonReporter < Reporter
+    def report_lints
+      grouped = lints.group_by(&:filename)
+      report = {
+        :metadata => {
+          :hamllint_version => VERSION,
+          :ruby_engine      => RUBY_ENGINE,
+          :ruby_patchlevel  => RUBY_PATCHLEVEL.to_s,
+          :ruby_platform    => RUBY_PLATFORM
+        },
+        :files    => grouped.map { |l| map_file(l) },
+        :summary  => {
+          :offense_count        => lints.length,
+          :target_file_count    => grouped.length,
+          :inspected_file_count => files.length
```

Comment:
Sorry, I missed this tiny use of `files`, so my other comments aren't quite valid.

How valuable do you think this information is?